### PR TITLE
Table layout utilities classes also apply to descendant elements; add table layout example docs

### LINKED ISSUE #2374: Tables probably shouldn't default to fixed width
I'm not sure if this is baked in to the style or part of the implementation where we use it, but using `table-layout: fixed;` doesn't produce great results for all but the simplest of tables (and even then has limitations).

Using a fixed width for the columns:

 - doesn't aid legibility
 - often causes table data to run over more lines than it should
 - is inefficient at using space (especially in reactive)
 - in extreme cases causes avoidable rendering errors

An example:

tables-001

This table pretty much illustrates all of the problems mentioned above.  However, with standard, flexible column widths it becomes more readable, more efficient and less broken

tables-002

Yes, we can still try harder to make things in tables less wordy, but still a huge improvement

### PR DESCRIPTION
**Pull request by jmuzina**
## Done

- Updated the `u-table-layout--auto` and `u-table-layout--fixed` utility classes to apply to nested / descended `table` elements as well as elements with those classes directly applied.
- Added example docs for the table layout utility settings. This includes:
  - Default (fixed) table layout (`$table-layout-fixed: true` and no utility class applied directly to a table)
  - Automatic table layout (`$table-layout-fixed: true` and `u-table-layout--auto` applied directly to a table)
  - Automatic table layout, nested (`$table-layout-fixed: true` and `u-table-layout--auto` applied to an ancestor element of a table 

Fixes #2374 , [WD-10925](https://warthogs.atlassian.net/browse/WD-10925) 

## QA

- Open [demo](https://vanilla-framework-5070.demos.haus/docs/settings/table-layout)
- Verify that the example tables in the table layout docs page appear as they do in the attached screenshots.
    - Tables in the table layout utility documentation page should have `table-layout: auto` applied.
- Review updated documentation:
  - Docs -> Settings -> Table layout

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots
image

[WD-10925]: https://warthogs.atlassian.net/browse/WD-10925?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jmuzina** [comment]
@lyubomir-popov Suggested that this change may be scoped a bit too large. Allowing the css to apply to all tables descended from the utility class element, as Bartek suggested in the Jira card, may be a bit too wide-reaching as this would allow, for example, applying the utility class at the `<body>`, level and making every `<table>` in the application to use `table-layout: auto`. 

Perhaps we should revise this to only apply to direct child elements, like so:

```scss
  @if ($table-layout-fixed) {
    .u-table-layout--auto {
      &,
      & > table {
        table-layout: auto !important;
      }
    }
  } @else {
    .u-table-layout--fixed {
      &,
      & > table {
        table-layout: fixed !important;
      }
    }
  }
```

Will review this with @bartaz upon his return from vacation.

File: scss/_utilities_layout.scss [hunk 1]
```diff
@@ -3,13 +3,21 @@
     @extend %fixed-width-container;
   }
 
+  // stylelint-disable selector-max-type -- table elements can use selector types
   @if ($table-layout-fixed) {
```
Comment (bartaz) [inline scss/_utilities_layout.scss:7]: While we are updating this, I'd remove the if statement. There is no huge benefit of having only one of those utils in the compiled version of Vanilla (they are tiny and will compress well). And it's just confusing for usage and documentation to have only one of the two available.

So let's just always have both utils available regardless of what is the default.

File: templates/docs/examples/utilities/table-layout/table-layout-default.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{% extends "_layouts/examples.html" %}
```
Comment (bartaz) [inline templates/docs/examples/utilities/table-layout/table-layout-default.html:1]: We have enough examples of default tables. I don't think there is a need to this one.

Comment (jmuzina) [reply templates/docs/examples/utilities/table-layout/table-layout-default.html]: Got it; removed that example from the PR

File: scss/_utilities_layout.scss (hunk 1 above)
Comment (jmuzina) [reply scss/_utilities_layout.scss]: If statement removed from PR and documentation updated accordingly

File: templates/docs/settings/table-layout.md [hunk 3]
```diff
@@ -8,7 +8,17 @@ By default, tables in Vanilla use `table-layout: fixed`.
 
 There are cases where you might want to use `table-layout: auto` - e.g. in automatically generated tables. To do this, you need to set the `$table-layout-fixed` variable to `true`.
 
-Depending on the value of `$table-layout-fixed`, a utility class is generated to allow overriding if necessary:
+Two utility classes are generated to allow overriding if necessary:
 
-- If `$table-layout-fixed` is set to `true` (the default), you can use `u-table-layout--auto` to override it.
-- if `$table-layout-fixed` is set to `auto` it adds a utility called `u-table-layout--fixed`.
+- `u-table-layout--auto` sets a table and its descendant tables to `table-layout: auto`.
```
Comment (bartaz) [inline templates/docs/settings/table-layout.md:13]: We don't want people to nest tables, so "its descendant tables" may be a bit misleading, so we should say that this utility sets "the table or any descendant tables of the element it's put on" to auto layout.

File: scss/standalone/settings_table-layout.scss [hunk 4]
```diff
@@ -0,0 +1,4 @@
+@import '../vanilla';
```
[the remark this answers is not in the document]
Comment (jmuzina) [reply scss/standalone/settings_table-layout.scss]: Removed file & references to it

File: templates/docs/settings/table-layout.md (hunk 3 above)
Comment (jmuzina) [reply templates/docs/settings/table-layout.md]: Updated documentation language accordingly

**Review by bartaz — APPROVED** [review]
LGTM, thanks

**Pull request merged**